jobbergateapi2/apps/applications/routers.py

```diff
@@ -65,5 +67,84 @@ async def applications_create(
     return application
 
 
+@router.delete(
+    "/applications/{application_id}", status_code=204, description="Endpoint to delete application"
+)
+async def application_delete(
+    current_user: User = Depends(get_current_user),
+    application_id: int = Query(..., description="id of the application to delete"),
+):
+    """
+    Given the id of an application, delete it from the database
+    """
+    s3_client = boto3.client("s3")
+    where_stmt = (applications_table.c.id == application_id) & (
+        applications_table.c.application_owner_id == current_user.id
+    )
+    get_query = applications_table.select().where(where_stmt)
+    raw_application = await database.fetch_one(get_query)
+    if not raw_application:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"Application {application_id=} not found for the user with id={current_user.id}",
+        )
+    application = Application.parse_obj(raw_application)
+    delete_query = applications_table.delete().where(where_stmt)
+    await database.execute(delete_query)
+    application_location = (
+        f"{settings.S3_BASE_PATH}/TEST/applications/{application.id}/jobbergate.tar.gz"
+        # f"{S3_BASE_PATH}/{application.owner_id}/applications/{application.id}/jobbergate.tar.gz"
```

Comment:
This is here for testing only, and the line below will be the one used when released